### PR TITLE
Updates for xbmc binding

### PR DESCRIPTION
**Pull request by giig1967g**
Hi,
sending you updates for XBMC binding.
Hope you will find useful
Added commands using Kodi's JSON-RPC:
- PLAYER.GETPROPERTIES: http://kodi.wiki/view/JSON-RPC_API/v6#Player.GetProperties
- XBMC.GETINFOLABEL: http://kodi.wiki/view/JSON-RPC_API/v6#XBMC.GetInfoLabels
- INPUT methods: http://kodi.wiki/view/JSON-RPC_API/v6#Input

Added also a custom refresh interval.

### CODE REVIEW

File: distribution/openhabhome/configurations/openhab_default.cfg [hunk 1]
```diff
@@ -2482,12 +2482,15 @@ tcp:refreshinterval=250
 # Port number for the web socket service (optional, defaults to 9090). Example:
 #xbmc:livingRoom.wsPort=9090
 
-# Username to connect to XBMC. (optional, defaults to none). Example:
+# Username to connect to XBMC. (optional, defaults to xbmc). Example:
```
Comment (sumnerboy12) [inline distribution/openhabhome/configurations/openhab_default.cfg]: so does the binding now default to authenticating using `xbmc:xbmc` - this sounds like breaking changes for anyone using the default configuration - why was this necessary?

**Comment by sumnerboy12** [comment]
I am not sure how you have created these changes but it looks like somehow every file in the binding has been removed and re-added - meaning it is impossible to determine what changes have actually been made. Any ideas how this might have happened? It is going to be impossible to review this change in its current state unfortunately.

**Comment by giig1967g** [comment]
Hi,
thanks for your reply.

It's the first time I use github, I might have done something wrong.
This is what I did:
I downloaded the files using Eclipse, then I made the changes to the files.
Then I opened a fork and uploaded the files.

I can do it in a different manner if you explain me how to do it :)

Regarding the default username and password, I noted that in the original
code they were set to 'xbmc' and I didn't change it.
I just updated the configuration file comment to reflect what was in the
original code.
You can check it by yourself in the file *internal\xbmcHosts.java*:












*public class XbmcHost {    private String hostname = "127.0.0.1";
private String username = "xbmc";    private String password = "xbmc";
private int rsPort = 8080;    private int wsPort = 9090;*
stefano

**Comment by sumnerboy12** [comment]
I am not sure what has gone wrong but you should only see the changes you made to the files when doing a diff in git, not every single file in the binding being removed and re-added. Sorry I am not sure what has gone wrong or how to resolve - try Googling perhaps?

As for the username/password, you are quite correct, apologies!

**Comment by giig1967g** [comment]
Hi sumnerboy12,
in order to solve the issue, should I refork the master or can I rollback my fork to the original state?
thanks

**Comment by giig1967g** [comment]
Hi Ben,
I will delete my reposotory and retry.
ok?

stefano

**Comment by giig1967g** [comment]
Hi Ben,
I have recreated the repository and added the files.
They now appear with the changes.

Can you confirm that I did the right procedure this time?

thanks
stefano

**Comment by sumnerboy12** [comment]
It looks like it has happened again.

If you look in the *Files Changed* tab of your Github pull request, the 
second file listed is XbmcActivator.java. It doesn't look like you made 
any changes to that file but Github is reporting the old file contents 
have been deleted and replaced with exactly the same contents as before.

Really sorry, not sure what is going on. Perhaps it is something to do 
with how you are editing the files, adding some non-standard line 
terminators perhaps?

**Comment by giig1967g** [comment]
Hi Ben,
I think you are looking inside the old fork that I have deleted, or I
thought I had deleted...

Here is my latest commit:
https://github.com/giig1967g/openhab/commit/80052fcac5328ee57de2136a17803506032e3a02
in there I can see the single changes.
I did it with smartGit v8.

Can you please have a look and tell me if it is fine?

thanks for your patience.
stefano

**Comment by sumnerboy12** [comment]
Yes - sorry - I just clicked the link on the previous email!

That is looking much better - I will try and review this evening when I 
have some free minutes.

Thanks.

**Comment by giig1967g** [comment]
Hi,
this is the pull request:
https://github.com/openhab/openhab/pull/4622

stefano

**Comment by watou** [comment]
Hi @giig1967g, just to confirm that this PR is obsolete and the changes you wanted were merged with #4622?

**Comment by giig1967g** [comment]
Hi John,
this PR is obsolete.
#4622 is the latest one

stefano

**Comment by watou** [comment]
Thanks very much, Stefano.
